src/WebJobs.Script/Binding/GeneralScriptBindingProvider.cs

```diff
@@ -0,0 +1,136 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.ObjectModel;
+using System.IO;
+using System.Reflection;
+using Microsoft.Azure.WebJobs.Host;
+using Microsoft.Azure.WebJobs.Script.Description;
+using Microsoft.Azure.WebJobs.Script.Extensibility;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Script.Binding
+{
+    // General binder that works directly off SDK interfaces. 
+    class GeneralScriptBindingProvider : ScriptBindingProvider
+    {
+        public ITooling Tooling { get; set; }
```

Comment:
Not quite. The ctor signature is locked. But I can move this to a FinishInit() method.